### ISSUE TITLE
Provide `match_type` dropdown in compare proxy creation modal window

### ISSUE THREAD

**Issue by cmutel** [cauldron/bw_matchbox#52]
<img width="1221" alt="Screenshot 2023-08-06 at 14 03 10" src="https://github.com/cauldron/bw_matchbox/assets/645217/4b6de4ad-c4de-4b54-95d7-849a5eb2590f">

There needs to be a dropdown selector somewhere in this modal, probably in the same line as the "create proxy process" window, which allows the selection of match types. The match types themselves were added to the `sharedData` in `compare.html` here: https://github.com/cauldron/bw_matchbox/blob/main/bw_matchbox/assets/templates/compare.html#L67

They have the form (but this is not fixed, more could be added:

```javascript
{
  '1': 'No direct match available',
  '2': 'Direct match with near-identical data', 
  '3': 'Match with updated data',
}
``` 

This should be translated into something like:

```html
<select name="match_type_select" id="match_type">
  <option value="1">No direct match available</option>
  <option value="2">Direct match with near-identical data</option>
  <option value="3">Match with updated data</option>
</select>
```

Name and id can be adjusted as needed.

The default value should be '3', `Match with updated data`. 

The `compare` endpoint will be afterwards be updated to store this information.

**Comment by cmutel** [comment]
`match_type` should be retrieved and added here: https://github.com/cauldron/bw_matchbox/blob/3134c85979c6c7749e9e7477bcc5c1677e78e575/bw_matchbox/assets/js/Compare/CompareCore.js#L347

**Comment by cmutel** [comment]
I added `match_type` to the JSON POST in 1ea2566942f7a1bd8aa9f709dd6193e8e9fab586

**Comment by lilliputten** [comment]
I think that would be nice to specify default value here implicitly, eg in `match_types_default`.

I don't understand why we need to show this field only at first displaying on the dialog ('When this window is first shown...')?

**Comment by cmutel** [comment]
> I think that would be nice to specify the default value here explicitly, eg in match_types_default.

44ac005165d2b00c64cc3738cdc3f4d435780f31

> I don't understand why we need to show this field only at first displaying on the dialog ('When this window is first shown...')?

Sorry, I didn't explain clearly. Edited to "There needs to be a dropdown selector somewhere in this modal", it is always present and changable. It can retain state if the user closes the modal and re-opens it, but this is not a requirement.

**Comment by lilliputten** [comment]
Fixed in PR #54.
